File: ekz_view/src/features/choice/models/commands/updateChoice.ts
import { gql } from '@apollo/client';
import { useAddChoiceMutation } from '../../../../graphql/generated/graphql';

export const UPDATE_CHOICE = gql`
  mutation updateChoice(
    $id: Int!
    $name: String!
    $url: String
    $description: String
    $evaluation: Int
  ) {
    updateChoice(
      input: {
        id: $id
        name: $name
        url: $url
        description: $description
        evaluation: $evaluation
      }
    ) {
      choice {
        ...Choice
      }
    }
  }
`;

export { useAddChoiceMutation };

File: ekz_api/app/graphql/mutations/choice/update_choice.rb
module Mutations::Choice
  class UpdateChoice < ::Mutations::BaseMutation
    argument :id, Int, required: true
    argument :name, String, required: true
    argument :url, String, required: false
    argument :description, String, required: false
    argument :evaluation, Int, required: false

    field :choice, Types::Choice::ChoiceType, null: false

    def resolve(id:, name:, url: nil, description: nil, evaluation: nil)
      choice = Bussiness::Choice::Command::UpdateCommand.call(
        id: id,
        name: name,
        url: url,
        description: description,
        evaluation: evaluation,
        )
      {
        choice: {
          id: choice.id,
          name: choice.name,
          url: choice.url,
          description: choice.description,
          evaluation: choice.evaluation,
          theme_id: choice.theme_id,
        }
      }
    end
  end
end